gdpr/gdpr.go

```diff
@@ -24,45 +21,22 @@ type Permissions interface {
 	// Determines whether or not to send PI information to a bidder, or mask it out.
 	//
 	// If the consent string was nonsensical, the returned error will be an ErrorMalformedConsent.
-	AuctionActivitiesAllowed(ctx context.Context, bidderCoreName openrtb_ext.BidderName, bidder openrtb_ext.BidderName, PublisherID string, gdprSignal Signal, consent string, weakVendorEnforcement bool, aliasGVLIDs map[string]uint16) (allowBidReq bool, passGeo bool, passID bool, err error)
+	AuctionActivitiesAllowed(ctx context.Context, bidderCoreName openrtb_ext.BidderName, bidder openrtb_ext.BidderName, PublisherID string, gdprSignal Signal, consent string, aliasGVLIDs map[string]uint16) (allowBidReq bool, passGeo bool, passID bool, err error)
 }
 
-// Versions of the GDPR TCF technical specification.
-const (
-	tcf2SpecVersion uint8 = 2
-)
-
 // NewPermissions gets an instance of the Permissions for use elsewhere in the project.
-func NewPermissions(ctx context.Context, cfg config.GDPR, vendorIDs map[openrtb_ext.BidderName]uint16, client *http.Client) Permissions {
+func NewPermissions(cfg config.GDPR, tcf2Config TCF2ConfigReader, vendorIDs map[openrtb_ext.BidderName]uint16, fetcher VendorListFetcher) Permissions {
 	if !cfg.Enabled {
 		return &AlwaysAllow{}
 	}
 
-	gdprDefaultValue := SignalYes
-	if cfg.DefaultValue == "0" {
-		gdprDefaultValue = SignalNo
-	}
```

Comment:
I decided to just use the string value rather than convert it to a `Signal` constant so we can continue making use of `SignalNormalize` in `impl.go` even though I removed the `config.GDPR` instance from the permissions object.